src/outputs/output_utils.hpp

```diff
@@ -96,11 +159,22 @@ struct VarInfo {
     }
   }
 
-  explicit VarInfo(const std::shared_ptr<Variable<Real>> &var)
+  explicit VarInfo(const std::shared_ptr<Variable<Real>> &var,
+                   const IndexShape &cellbounds)
       : VarInfo(var->label(), var->metadata().getComponentLabels(), var->NumComponents(),
-                var->GetDim(6), var->GetDim(5), var->GetDim(4), var->GetDim(3),
-                var->GetDim(2), var->GetDim(1), var->metadata(), var->IsSparse(),
-                var->IsSet(Metadata::Vector)) {}
+                var->GetDim(), var->metadata(), var->GetTopologicalElements(),
+                var->IsSparse(), var->IsSet(Metadata::Vector), cellbounds) {}
+
+  static std::vector<VarInfo> GetAll(const VariableVector<Real> &vars,
+                                     const IndexShape &cellbounds);
+
+  bool operator==(const std::string &other) const { return other == label; }
+
+ private:
+  // TODO(JMM): Probably nx_ and rnx_ both not necessary... but it was
+  // easiest for me to reason about it this way.
+  std::array<int, VNDIM> nx_;
+  std::vector<int> rnx_;
```

Comment:
This is just the limitations of my ability to reason about index gymnastics. I think this could be simplified but I'm not sure it's worthwhile to do so.